docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,105 @@
+Therefore, later we will provide other isolation methods as an extension to NP.
+
+### Advantages:
+- does not require any changes in modules implementations, can be implemented in the Fybrik control plane only.
+- can be combined with other isolation methods
+
+### Disadvantages:
+- complicated in cross clusters deployments.
+- NP are implemented by the network plugin. Creating a NetworkPolicy resource without a controller - a Container Network 
+Interface (CNI) that implements it will have no effect. Default Kind deployment which by default uses "kindnetd” doesn’t 
+support NP. It requires installation another CNI, e.g. [Calico](https://github.com/projectcalico/calico). Here is the 
+[instructions](https://alexbrand.dev/post/creating-a-kind-cluster-with-calico-networking/). All K8s clusters in 
+production deployments use CNIs which support NP.
+- requires information about workload deployment, and/or setting labels to the workload
+
+## Implementation
+
+The “from” element of ingress 
+[NetworkPolicySpec](https://kubernetes.io/docs/reference/kubernetes-api/policy-resources/network-policy-v1/#NetworkPolicySpec) 
+allows to restrict incoming network traffic based on “podSelectors” (labels based), and/or 
+“namespaceSelectors” (labels based) and/or “ipBlocks”. The last one permits incoming network traffic from predefined IP 
+address blocks.
+
+Current “FybrikApplicationSpec” has the “[selector](https://fybrik.io/v1.2/reference/crds/#fybrikapplicationspecselector)” 
+element, which is a combination of “clusterName” and “workloadSelector”. Unfortunately, based on this information we can 
+implement only one label based “podSelector”. Furthermore, for the backward compatibility of the current implementation 
+assumes that a Read scenario should have at least one label, which is not always true. 
+
+### Suggetions:
+
+- extend the “FybrikApplicationSpec.selector” element with an array of 
+[NetworkPolicyPeers](https://github.com/kubernetes/api/blob/59fcd23597fd090dba6b7e903eb0a8c9e8efb0a6/networking/v1/types.go#L183)
+this “NetworkPoliciesPeers” array will be an input element for the NP `from`element. This will restrict possible incoming
+connections to the relevant Fybrik module.
+- a blueprint will provide the ingress input data for chain connected modules.
+
+- deprecate the “workloadSelector” element, but continue to support it for backward compatibility. Process it as an 
+additional “podSelector” entry in the “NetworkPoliciesPeers” array
+
+- terminate the backward compatibility assumption that all Read flow should have at least a single workload label. 
+It was deprecated several Fybrik versions ago.
+
+- change the Flow field to be required instead of optional.
+
+- if the NetworkPolicyPeers array and “workloadSelector” are empty, no Network Policies will be created, which is 
+similar to current implementation.
+
+- extend blueprint’s module install/uninstall process by installation of a relevant instance of network policies.
```

Comment:
So network policies will be defined by the plotter controller or by the blueprint controller?

Reply:
the blueprint controller

Reply:
Wouldn't the plotter define them, and the blueprint deploy them?

Reply:
The plotter defines the data plane so yes it defines who will talk with whom, but it is semantic.

Reply:
So the plotter defines, and the blueprint deploys.  Right?

Reply:
Yes.